### PR TITLE
Fixed possible out-of-bound access in circles drawing

### PR DESCRIPTION
**Pull request by asmorkalov**
### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [ ] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: modules/imgproc/src/drawing.cpp [hunk 1]
```diff
@@ -1579,7 +1579,7 @@ Circle( Mat& img, Point center, int radius, const void* color, int fill )
                     ICV_HLINE( tptr, x11, x12, color, pix_size );
             }
 
-            if( (unsigned)y12 < (unsigned)size.height )
+            if( y12 >= 0 && y12 < size.height )
```
Comment (vrabaud) [inline modules/imgproc/src/drawing.cpp:1582]: Though standard C++ promotion happens, y12 and size.height have different types so the compiler might complain. Casting to (int64_t) will it should be fine with our CI.

File: modules/imgproc/test/test_drawing.cpp [hunk 2]
```diff
@@ -913,4 +913,11 @@ INSTANTIATE_TEST_CASE_P(
     )
 );
 
+TEST(Drawing, DISABLED_circle_overflow)
```
Comment (opencv-alalek) [inline modules/imgproc/test/test_drawing.cpp:916]: Why `DISABLED_`?

Comment (asmorkalov) [reply modules/imgproc/test/test_drawing.cpp:916]: It takes significant time. I decided to disable it by default. Should I return int back?

Comment (opencv-alalek) [reply modules/imgproc/test/test_drawing.cpp:916]: `applyTestTag(CV_TEST_TAG_VERYLONG);` or at least add comment.

Comment (asmorkalov) [reply modules/imgproc/test/test_drawing.cpp:916]: Enabled test with `CV_TEST_TAG_VERYLONG` option.

**Comment by asmorkalov** [comment]
@vrabaud @opencv-alalek firendly reminder.

**Comment by vrabaud** [comment]
👍

**Review by vrabaud — APPROVED** [review]

**Pull request merged**